### PR TITLE
add webhooks for draftAndPublish + sanitize all webhooks info

### PR DESCRIPTION
**Pull request by petersg83**
### Description of what you did:
- Triggers webhooks when publishing/unpublishing
- sanitize the webhooks contents (breaking change)
- fix wrong creator fields in media webhooks

### CODE REVIEW

File: docs/v3.x/concepts/webhooks.md [hunk 1]
```diff
@@ -54,16 +54,25 @@ You can delete a webhook by clicking on the `trash` icon.
...
-| `entry.update` | Triggered when a Content Type entry is updated. |
-| `entry.delete` | Triggered when a Content Type entry is deleted. |
-| `media.create` | Triggered when a media is created.              |
-| `media.delete` | Triggered when a media is deleted.              |
+| Name              | Description                                           |
+| ----------------- | ----------------------------------------------------- |
+| `entry.create`    | Triggered when a Content Type entry is created.       |
+| `entry.update`    | Triggered when a Content Type entry is updated.       |
+| `entry.delete`    | Triggered when a Content Type entry is deleted.       |
+| `entry.publish`   | Triggered when a Content Type entry is published.\*   |
```
Comment (Convly) [inline docs/v3.x/concepts/webhooks.md:62]: What's the purpose of these `\*`?

File: docs/v3.x/concepts/webhooks.md [hunk 2]
```diff
@@ -54,16 +54,25 @@ You can delete a webhook by clicking on the `trash` icon.
...
+| `media.create`    | Triggered when a media is created.                    |
+| `media.update`    | Triggered when a media is updated.                    |
+| `media.delete`    | Triggered when a media is deleted.                    |
+
+\*only when `draftAndPublish` is enabled on this Content Type.
 
 ## Payloads
 
+:::tip NOTE
+Private fields are not sent in the payload.
```
Comment (Convly) [inline docs/v3.x/concepts/webhooks.md:73]: and passwords*

File: packages/strapi-plugin-content-manager/services/ContentManager.js [hunk 3]
```diff
@@ -83,17 +83,34 @@ module.exports = {
...
-    return strapi.entityService.update(
+  async publish(params, model) {
+    const modelDef = strapi.getModel(model);
+
+    const publishedEntry = await strapi.entityService.update(
       { params, data: { [PUBLISHED_AT_ATTRIBUTE]: new Date().toISOString() } },
       { model }
     );
+
+    strapi.eventHub.emit('entry.publish', {
```
Comment (Convly) [inline packages/strapi-plugin-content-manager/services/ContentManager.js:94]: Little side-note: Shouldn't we have something like an enum used to store all the literal values of the webhooks? As the app (and the amount of webhook) grows, we (IMO) should consider grouping those kinds of values together to make contributions easier.
What do you think?

File: packages/strapi/lib/services/__tests__/entity-service.test.js [hunk 4]
```diff
@@ -88,6 +88,7 @@ describe('Entity service', () => {
         const fakeModel = {
           kind: 'contentType',
           modelName: 'test-model',
+          options: {},
```
Comment (Convly) [inline packages/strapi/lib/services/__tests__/entity-service.test.js:91]: Why do you need to add this here?

**Review by Convly — CHANGES_REQUESTED** [review]
Some questions

File: docs/v3.x/concepts/webhooks.md (hunk 1 above)
Comment (petersg83) [reply docs/v3.x/concepts/webhooks.md:62]: I don't know why there is the `\`, I only put `*`, some autoformatting changed that during the commit I think.
It's just an asterix :)

File: packages/strapi/lib/services/__tests__/entity-service.test.js (hunk 4 above)
Comment (petersg83) [reply packages/strapi/lib/services/__tests__/entity-service.test.js:91]: Because the test was broken "options is undefined"

**Review by Convly — APPROVED** [review]
LGTM, waiting for the tests

**Pull request merged**